src/shared/cpu-set-util.c

```diff
@@ -82,6 +82,23 @@ char *cpu_set_to_range_string(const CPUSet *set) {
         return TAKE_PTR(str) ?: strdup("");
 }
 
+char* cpu_set_to_mask_string(const CPUSet *a) {
+        _cleanup_free_ char *str = NULL;
+        unsigned mask = 0;
+        int i, r;
+
+        for (i = 0; (size_t) i < a->allocated * 8; i++) {
+                if (!CPU_ISSET_S(i, a->allocated, a->set))
+                        continue;
+
+                mask += 1 << i;
```

Comment:
this is a bit icky. first of all i think |= would make more sense here, if you are putting together a bitmask. but the other thing is that `1` written like this is a 32bit signed integer, so once you hit i == 31 things get weird... should really be 1U...

but then again, i think the type is  generally not right, see above.